src/rgw/rgw_sal_motr.cc

```diff
@@ -561,7 +564,7 @@ int MotrBucket::list(const DoutPrefixProvider *dpp, ListParams& params, int max,
     ocount++;
     if (ocount == max) {
         is_truncated = true;
-        results.next_marker = key_vec[max - 1] + "\x00";
+        results.next_marker = keys[max - 1] + " ";
```

Comment:
hmm

Reply:
Yes, we need to look up for the next keys in the next call. If we use the last one in the next lookup - we'll have it duplicated in the result.